### PR TITLE
test(ot3): add Hypothesis motion test

### PR DESCRIPTION
**Pull request by ahiuchingau**
We're now using [Hypothesis](https://hypothesis.readthedocs.io/en/latest/) to test our motion plan!

After testing some edge cases, I've made some changes to a few util functions to work better with floats values.

### CODE REVIEW

File: hardware/tests/opentrons_hardware/hardware_control/test_motion_plan.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    formatted: Iterator[np.float64] = (np.float64(i) for i in coord)
+    return Coordinates.from_iter(formatted)
+
+
+def reject_close_coordinates(a: Coordinates, b: Coordinates) -> bool:
+    """Reject example if the coordinates are too close.
+
+    Consecutive coordinates must be at least 1mm apart in one of the axes.
+    """
+    return any(abs(b.vectorize() - a.vectorize()) > 1.0)
```
Comment (sfoster1) [inline hardware/tests/opentrons_hardware/hardware_control/test_motion_plan.py:62]: IMO let's make this `> 0.05` or something - we'll generate moves like that sometimes, when doing very small moves in calibration

Comment (sfoster1) [reply hardware/tests/opentrons_hardware/hardware_control/test_motion_plan.py:62]: or have a second test of like `generate_close_coordinates` that only does coordinates that differ by `0.01 < difference < 1.0`

File: hardware/opentrons_hardware/hardware_control/motion_planning/move_utils.py [hunk 2]
```diff
@@ -24,6 +24,11 @@ def apply_constraint(constraint: np.float64, input: np.float64) -> np.float64:
     return np.copysign(np.minimum(abs(constraint), abs(input)), input)
 
 
+def check_against_constraint(constraint: np.float64, input: np.float64) -> bool:
```
Comment (sfoster1) [inline hardware/opentrons_hardware/hardware_control/motion_planning/move_utils.py]: this feels a little generically named for what it does, maybe `check_less_or_close` is a little more descriptive? I do think this is the right approach, though.

File: hardware/opentrons_hardware/hardware_control/motion_planning/move_utils.py [hunk 3]
```diff
@@ -66,7 +71,7 @@ def targets_to_moves(initial: Coordinates, targets: List[MoveTarget]) -> Iterato
                 ),
             ),
         )
-        log.debug(f"Built move from {initial} to {target} as {m}")
+        log.info(f"Built move from {initial} to {target} as {m}")
```
Comment (sfoster1) [inline hardware/opentrons_hardware/hardware_control/motion_planning/move_utils.py]: these feel a little wordy to be `info`

**Review by sfoster1 — APPROVED** [review]
Functionality looks good and tests look very cool! Would love to change some names and log levels but the functionality looks great.
